### PR TITLE
feat: implement regr_slope, regr_intercept, regr_r2, regr_sxx, regr_syy, regr_sxy aggregates

### LINKED ISSUE #4552: Natively support regr_sxx, regr_syy, regr_sxy, regr_slope, regr_intercept, regr_r2 aggregates
## Background

Spark provides the SQL standard linear-regression aggregate functions `regr_*(y, x)`, which compute single-pass statistics over rows where both `y` and `x` are non-null. They are standard descriptive statistics (same set as PostgreSQL), not ML.

SQL file tests added in #4551 (`spark/src/test/resources/sql-tests/expressions/aggregate/regr.sql`) establish the current state empirically:

- **Already accelerated natively:** `regr_count`, `regr_avgx`, `regr_avgy`. Spark implements these as `RuntimeReplaceableAggregate`s that lower to `Count` / `Average`, which Comet already supports, so they run without any new code.
- **Currently fall back to Spark (this issue):** `regr_sxx`, `regr_syy`, `regr_sxy`, `regr_slope`, `regr_intercept`, `regr_r2`. In #4551 these are covered with `query spark_answer_only` (correctness only).

## Proposal

Add native Comet support for the six functions that currently fall back. None of these are greenfield: they all build on the streaming moment accumulators Comet already implements for `covar_pop`, `var_pop`, and `corr`. From Spark's `linearRegression.scala`:

- `regr_sxy` extends `Covariance` (same accumulator as Comet's `covar_pop`/`covar_samp`), with a different final expression.
- `regr_r2` extends `PearsonCorrelation` (same accumulator as Comet's `corr`), with a different final expression.
- `regr_slope` and `regr_intercept` are `DeclarativeAggregate`s composing `CovPopulation` + `VariancePop`, with a null-pair guard on the variance update.
- `regr_sxx` and `regr_syy` are `RuntimeReplaceableAggregate`s that lower to an internal `RegrReplacement` declarative aggregate (count times variance over the non-null pairs).

So the work is to wire these aggregate classes through `QueryPlanSerde` / the aggregate serde and expose the appropriate final expressions over the existing native accumulators, plus match Spark's null-pair filtering semantics exactly.

## Acceptance criteria

- The six functions execute natively in Comet and match Spark.
- Update `expressions/aggregate/regr.sql` (from #4551) to switch these queries from `query spark_answer_only` back to the default `query` mode so native execution is asserted.

## Notes

These could be tackled incrementally (for example `regr_sxy` and `regr_r2` first, since they map most directly onto the existing covariance and correlation accumulators).

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes #4552.

## Rationale for this change

Comet already accelerates `regr_avgx`, `regr_avgy`, and `regr_count` (Spark rewrites those to `Average`/`Count`), but the remaining six SQL-standard linear regression aggregates fell back to Spark. This PR implements native support for `regr_slope`, `regr_intercept`, `regr_r2`, `regr_sxx`, `regr_syy`, and `regr_sxy`, so a query using any of them can run fully on Comet instead of falling back.

## What changes are included in this PR?

- A new native `Regr` aggregate UDF in `native/spark-expr/src/agg_funcs/regr.rs`. Rather than re-implementing the statistics, each function is composed from Comet's existing Spark-compatible `CovarianceAccumulator` and `VarianceAccumulator`. This keeps the partial aggregation state byte-compatible with the buffer layout Spark's planner declares for the partial to final shuffle:
  - `regr_sxx` / `regr_syy` reach Comet as `RegrReplacement` (a `CentralMomentAgg`, 3-field buffer) and reuse the variance accumulator, evaluating to `m2`.
  - `regr_sxy` (a population `Covariance`, 4-field buffer) reuses the covariance accumulator, evaluating to the co-moment `ck`.
  - `regr_r2` (a `PearsonCorrelation`, 6-field buffer) composes covariance + two variances.
  - `regr_slope` / `regr_intercept` (a declarative composite, 7-field buffer) compose population covariance + variance.
- `regr_r2` matches Spark's behavior of returning `1.0` when the dependent variable is constant but the independent variable varies (a perfect horizontal fit). This is the one case where Spark and DataFusion's `regr_r2` diverge (DataFusion returns `null`), so a Comet-specific accumulator is warranted.
- A new `Regr` protobuf message (with a `RegrType` enum) wired through `QueryPlanSerde` and the native `PhysicalPlanner`.
- Scala serde for `RegrSlope`, `RegrIntercept`, `RegrR2`, `RegrSXY`, and `RegrReplacement` (the rewrite target for `regr_sxx`/`regr_syy`).
- Updated the expression support status in the user guide.

This work was scaffolded using the `implement-comet-expression` project skill.

## How are these changes tested?

- Expanded the existing `spark/src/test/resources/sql-tests/expressions/aggregate/regr.sql` so all six functions are verified to run natively and match Spark (previously `spark_answer_only`). New cases cover NULL pairs, single-pair input, constant independent variable (slope/intercept/r2 to NULL), constant dependent variable (r2 to 1.0), grouped aggregation, and literal/column argument mixes.
- Added Rust unit tests in `regr.rs` covering perfect-fit slope/intercept/r2, the constant-y and constant-x edge cases, single-pair and empty input, NULL-pair skipping, the raw moments, and partial-state merge across batches.

### CODE REVIEW

**Comment by manuzhang** [comment]
@andygrove Can you rebase this PR?

**Comment by andygrove** [comment]
> @andygrove Can you rebase this PR?

@manuzhang Done

File: native/proto/src/proto/expr.proto [hunk 1]
```diff
@@ -263,6 +264,36 @@ message Correlation {
...
+  // Only consulted for SLOPE and INTERCEPT. When true (Spark 3.5+), VariancePop(x)
+  // is computed only over rows where both y and x are non-null. When false
+  // (Spark 3.4), VariancePop(x) includes every row where x is non-null even if y
+  // is null, matching the pre-fix Spark 3.4 semantics.
+  bool filter_var_by_pair_nulls = 5;
+  // Only consulted for R2. Spark swapped the degenerate-case handling of
+  // regr_r2. When true, a constant dependent variable (m2(y) = 0) returns 1.0
+  // and a constant independent variable (m2(x) = 0) returns null. When false
+  // those two cases are reversed. The swap is present in the Spark versions
+  // Comet builds against for 3.5 and later (3.5.9, 4.0.3+, 4.1, 4.2) but not
```
Comment (manuzhang) [inline native/proto/src/proto/expr.proto:292]: The list can become style. This comment can be simplified to

```
// The swap is present in the Spark versions Comet builds against for 3.5+, but not in 3.4.
```

**Review by manuzhang — APPROVED** [review]
LGTM. Some nit comments.

File: spark/src/main/scala/org/apache/comet/serde/aggregates.scala [hunk 2]
```diff
@@ -763,6 +763,131 @@ object CometCorr extends CometAggregateExpressionSerde[Corr] {
...
+      builder.setRegrType(regrType)
+      builder.setDatatype(dataType.get)
+      // Spark 3.5 fixed regr_slope/regr_intercept so VariancePop(x) only counts
+      // rows where both y and x are non-null. Spark 3.4 counts every row where x
+      // is non-null. The native accumulator only consults this for slope/intercept.
+      builder.setFilterVarByPairNulls(isSpark35Plus)
+      // Spark swapped regr_r2's degenerate-case handling: a constant dependent
+      // variable now yields 1.0 (was null) and a constant independent variable
+      // yields null (was 1.0). The swap is present in the Spark versions Comet
+      // builds against for 3.5 and later (3.5.9, 4.0.3+, 4.1, 4.2) but not in 3.4.
```
Comment (manuzhang) [inline spark/src/main/scala/org/apache/comet/serde/aggregates.scala:796]: same here.

File: spark/src/main/scala/org/apache/comet/serde/aggregates.scala [hunk 3]
```diff
@@ -763,6 +763,131 @@ object CometCorr extends CometAggregateExpressionSerde[Corr] {
...
+    val child2Expr = exprToProto(x, inputs, binding)
+    val dataType = serializeDataType(DoubleType)
+
+    if (child1Expr.isDefined && child2Expr.isDefined && dataType.isDefined) {
+      val builder = ExprOuterClass.Regr.newBuilder()
+      builder.setChild1(child1Expr.get)
+      builder.setChild2(child2Expr.get)
+      builder.setRegrType(regrType)
+      builder.setDatatype(dataType.get)
+      // Spark 3.5 fixed regr_slope/regr_intercept so VariancePop(x) only counts
```
Comment (manuzhang) [inline spark/src/main/scala/org/apache/comet/serde/aggregates.scala:789]: Maybe better (describing what not how)

```
VariancePop(x) only counts rows where both y and x are non-null in Spark 3.5+, but counts every row where x is non-null in 3.4

```

**Comment by andygrove** [comment]
Merged. Thanks @manuzhang
